server/result/serializers.py

```diff
@@ -7,3 +10,27 @@ class Meta:
         model = ModelResult
         fields = "__all__"
         read_only_fields = ["owner", "approved_at", "approval_status"]
+
+    def validate(self, data):
+        benchmark = data["benchmark"]
+        mlcube = data["model"]
+        dataset = data["dataset"]
+        last_benchmarkmodel = (
+            BenchmarkModel.objects.filter(benchmark__id=benchmark.id, model_mlcube__id=mlcube.id)
+            .order_by("-created_at")
+            .first()
+        )
+        if not last_benchmarkmodel:
```

Comment:
Current implementation raises an error when the benchmark is executed with the reference model, since this model is not associated to the benchmark through the `BenchmarkModel` object.

Reply:
We would need an if in the lines of 
`if not last_benchmarkmodel and not referencemodel:`
I'm guessing this would require retrieving the current benchmark

Reply:
Do we upload results for reference model as well?

Reply:
Yes we do upload results for reference models

Reply:
Which dataset is used for this result upload?